staging/src/k8s.io/apimachinery/pkg/runtime/splice_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime_test
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestSpliceBuffer(t *testing.T) {
+	testBytes := []byte{0x01, 0x02, 0x03, 0x04}
+
+	testCases := []struct {
+		testname string
+		testFunc func(sb runtime.Splice, buf *bytes.Buffer)
+	}{
+		{
+			"Basic Write",
```

Comment:
nit: please use explicit field names, i.e.:
```
name: "Basic Write",
test: func(....) ...
```

it's useful for future extensions if someone wants to add a new field but set it only for subset of usecases